File: 17-trees3/1-intro/main.cpp
// Introduction to Trie and N-ary Tree Nodes - C++ Example

#include <iostream>
#include <unordered_map>
#include <vector>

struct TrieNode {
    std::unordered_map<char, TrieNode*> children;
    bool is_end_of_word;
    TrieNode() : is_end_of_word(false) {}
};

struct NaryTreeNode {
    int data;
    std::vector<NaryTreeNode*> children;
    NaryTreeNode(int val) : data(val) {}
};

int main() {
    TrieNode trie_root;
    NaryTreeNode nary_root(1);
    std::cout << "Trie and N-ary root nodes created." << std::endl;
    return 0;
}

File: 17-trees3/1-intro/main.java
// Introduction to Trie and N-ary Tree Nodes - Java Example

import java.util.HashMap;
import java.util.ArrayList;

class TrieNode {
    HashMap<Character, TrieNode> children = new HashMap<>();
    boolean isEndOfWord = false;
}

class NaryTreeNode {
    int data;
    ArrayList<NaryTreeNode> children = new ArrayList<>();
    NaryTreeNode(int data) {
        this.data = data;
    }
}

class Trees3Intro {
    public static void main(String[] args) {
        TrieNode trieRoot = new TrieNode();
        NaryTreeNode naryRoot = new NaryTreeNode(1);
        System.out.println("Trie and N-ary root nodes created.");
    }
}
